packages/lending/src/adapter.ts

```diff
@@ -0,0 +1,1051 @@
+        account,
+        logics: leverageShortlogics,
+      },
+      this.permitType
+    );
+
+    const buildRouterTransactionRequest = (
+      args?: Omit<apisdk.RouterData, 'chainId' | 'account' | 'logics'>,
+      apiKey?: string
+    ): Promise<common.TransactionRequest> =>
+      apisdk.buildRouterTransactionRequest(
+        { ...args, chainId: this.chainId, account, logics: leverageShortlogics },
+        apiKey ? { 'x-api-key': apiKey } : undefined
+      );
+
+    return {
+      fields: {
+        srcToken,
+        srcAmount,
+        destToken,
+        destAmount: supplyLogic.fields.input.amount,
+        portfolio,
+        afterPortfolio,
+      },
+      estimateResult,
+      buildRouterTransactionRequest,
+      logics: leverageShortlogics,
+    };
+  }
+
+  // 1. flashloan destToken
+  // 2. swap destToken to srcToken
+  // 3. repay srcToken
+  // 4. add fund destToken
+  // 5. withdraw destToken by aDestToken
+  // 6. flashloan repay destToken
+  // * srcToken => borrowToken, repayToken
+  // * destToken => depositToken, collateralToken
+  async getDeleverage(
+    protocolId: string,
+    marketId: string,
+    params: BaseParams,
+    account: string,
+    portfolio?: Portfolio
+  ): Promise<BaseFields> {
+    const { srcAmount } = params;
+    const srcToken = common.classifying(params.srcToken);
+    const destToken = common.classifying(params.destToken);
+    const wrappedSrcToken = srcToken.wrapped;
+    const wrappedDestToken = destToken.wrapped;
+
+    const deleveragelogics: apisdk.Logic<any>[] = [];
+    const protocol = this.getProtocol(protocolId);
+
+    portfolio = portfolio || (await protocol.getPortfolio(account, marketId));
+    const afterPortfolio = portfolio.clone();
+
+    // ---------- Pre-calc quotation ----------
+    const swapper = this.findSwapper([wrappedDestToken, wrappedSrcToken]);
+    //  get the quotation for how much collateral token is needed to exchange for the repay amount
+    const swapQuotation = await swapper.quote({
+      tokenIn: wrappedDestToken,
+      output: { token: wrappedSrcToken, amount: srcAmount },
+      slippage: defaultSlippage,
+    });
+
+    // TODO: seems we do not need this
+    // // convert swap type to exact in
+    // swapQuotation = await swapper.quote({
+    //   input: swapQuotation.input,
+    //   tokenOut: wrappedDestToken,
+    //   slippage: defaultSlippage,
+    // });
+
+    // ---------- flashloan ----------
+    const flashLoanAggregatorQuotation = await apisdk.protocols.utility.getFlashLoanAggregatorQuotation(this.chainId, {
+      loans: [swapQuotation.input],
+    });
+
+    const [flashLoanLoanLogic, flashLoanRepayLogic] = apisdk.protocols.utility.newFlashLoanAggregatorLogicPair(
+      flashLoanAggregatorQuotation.protocolId,
+      flashLoanAggregatorQuotation.loans.toArray()
+    );
+    deleveragelogics.push(flashLoanLoanLogic);
+
+    // ---------- swap ----------
+    const swapTokenLogic = swapper.newSwapTokenLogic(swapQuotation);
```

Comment:
need to check if srcToken == destToken. plz check other swaps too.

ref
![radiant-v2](https://github.com/dinngo/protocolink-js-sdk/assets/27182072/f09896e2-1c2a-48e2-8dee-0802e6efb7d0)

Reply:
@chouandy will add this check

Reply:
This [commit](https://github.com/dinngo/protocolink-js-sdk/pull/15/commits/6b9ffc28025e318803e6e29b86ff0328d71f577a#diff-b79db2de06cbe1ac07ca0c06c1ff25c20b3b0334c36d073027b973c17a8d802a) has added this check for leverageLong, leverageShort, deleverage.